File: Examples/Features/Worksheets/MoveWorksheet.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace GrapeCity.Documents.Excel.Examples.Features.Worksheets
{
    public class MoveWorksheet : ExampleBase
    {
        public override void Execute(GrapeCity.Documents.Excel.Workbook workbook)
        {
            //Load template file AgingReport.xlsx from resource
            var fileStream = this.GetResourceStream("xlsx\\AgingReport.xlsx");
            workbook.Open(fileStream);

            //Move the active sheet to the end of current workbook
            var move_worksheet = workbook.ActiveSheet.Move();
            move_worksheet.Name = "Move of " + workbook.ActiveSheet.Name;
        }

        public override string TemplateName
        {
            get
            {
                return "AgingReport.xlsx";
            }
        }

        public override string[] UsedResources
        {
            get
            {
                return new string[] { "xlsx\\AgingReport.xlsx" };
            }
        }
    }
}

File: Examples.VB/Features/Worksheets/MoveWorksheet.vb
Namespace Features.Worksheets
    Public Class MoveWorksheet
        Inherits ExampleBase

        Public Overrides Sub Execute(workbook As Excel.Workbook)
            'Load template file AgingReport.xlsx from resource
            Dim fileStream = GetResourceStream("AgingReport.xlsx")
            workbook.Open(fileStream)

            'Move the active sheet to the end of current workbook
            Dim move_worksheet = workbook.ActiveSheet.Move()
            move_worksheet.Name = "Move of " & workbook.ActiveSheet.Name
        End Sub

        Public Overrides ReadOnly Property TemplateName As String
            Get
                Return "AgingReport.xlsx"
            End Get
        End Property

        Public Overrides ReadOnly Property UsedResources As String()
            Get
                Return New String() {"xlsx\AgingReport.xlsx"}
            End Get
        End Property
    End Class
End Namespace
